### ISSUE TITLE
Google Map shortcode returns error & warning messages

### ISSUE THREAD

**Issue by stanalyst** [SparkDevNetwork/Rock#6122]
### Description

Tried to setup and use the googlemap shortcode in an HTML content block - and received the messages below in the console. I can't confirm if these impact functionality or performance - but it does seem like the shortcode should be updated to stop using depricated features and to follow best practices.

"Google Maps JavaScript API has been loaded directly without loading=async. This can result in suboptimal performance. For best-practice loading patterns please see https://goo.gle/js-api-loading"

"GET https://maps.googleapis.com/maps/api/mapsjs/gen_204?csp_test=true net::ERR_BLOCKED_BY_CLIENT"

"google.maps.event.addDomListener() is deprecated, use the standard addEventListener() method instead: https://developer.mozilla.org/docs/Web/API/EventTarget/addEventListener The feature will continue to work and there is no plan to decommission it."

"As of February 21st, 2024, google.maps.Marker is deprecated. Please use google.maps.marker.AdvancedMarkerElement instead. At this time, google.maps.Marker is not scheduled to be discontinued, but google.maps.marker.AdvancedMarkerElement is recommended over google.maps.Marker. While google.maps.Marker will continue to receive bug fixes for any major regressions, existing bugs in google.maps.Marker will not be addressed. At least 12 months notice will be given before support is discontinued. Please see https://developers.google.com/maps/deprecations for additional details and https://developers.google.com/maps/documentation/javascript/advanced-markers/migration for the migration guide."

### Actual Behavior

Using the googlemap shortcode on a page - error and warning messages are returned to the console.

### Expected Behavior

Using the googlemap shortcode on a page should not return error/warning messages.

### Steps to Reproduce

- put an HTML content block on a page
- add the following code to the HTML block
```
{[ googlemap ]}
    [[ marker location:'33.640705,-112.280198' title:'Spark Global Headquarters']]
        <strong>Spark Global Headquarters</strong>
        It's not as grand as it sounds.<br>
        <img src="https://rockrms.blob.core.windows.net/misc/spark-logo.png" width="179" height="47">          
    [[ endmarker ]]
    [[ marker location:'33.52764, -112.262571']][[ endmarker ]]
{[ endgooglemap ]}
```
- view the page in a web browser
- open the devtools console and note the messages

### Issue Confirmation

- [X] Perform a search on the [Github Issues](https://github.com/issues?q=is%3Aissue+user%3ASparkDevNetwork+-repo%3ARock) to see if your bug or enhancement is already reported.
- [X] Reproduced the problem on a fresh install or on the [demo site](https://rock.rocksolidchurchdemo.com/).

### Rock Version

1.16.6.9

### Client Culture Setting

en-US

**Comment by chead4** [comment]
@stanalyst Hi Stan - Thanks for reporting the issue! I’m happy to let you know that it’s been resolved and the fix is included in version 17.0. I’ve tested it in the [pre-alpha](https://prealpha.rocksolidchurchdemo.com/intranet/shared-documents#) environment, and it’s working as expected. Commit [983d006 ](https://github.com/SparkDevNetwork/Rock/commit/983d00658e24b9fbe807a79e19bef54e5dbbeeb5).

**Comment by nlabarbera** [comment]
Hey @stanalyst, here’s a bit more context: We resolved the deprecation message in v17 by introducing the new Maps ID Global Attribute. With this update, you can leverage the latest Google Maps API. In your Google Cloud Console, simply create a Map ID and use that value in Global Attributes. We're still working on detailed documentation for this, so stay tuned!

I'll have the team address the async loading and optimize the addEventListener() functionality to resolve those warnings effectively.
